lib/skate_web/controllers/route_tabs_controller.ex

```diff
@@ -5,7 +5,10 @@ defmodule SkateWeb.RouteTabsController do
   alias Skate.Settings.RouteTab
 
   def update(conn, %{"route_tabs" => route_tabs} = _params) do
-    username = AuthManager.Plug.current_resource(conn)
+    username =
+      conn
+      |> AuthManager.Plug.current_resource()
+      |> AuthManager.username_from_resource()
```

Comment:
**suggestion:** In this and a few other places where we do get username, it's then passed into functions that use it to get / update some relevant data from the database - route tabs in this instance. However, for most of these we should just be able to rework the underlying function to take a user ID instead and use that in the relevant queries. Making that change might be too much added complexity for this ticket but it would be good to create a followup tech cleanup ticket for it.

Reply:
💯 - I was worried about putting that all into this PR, but have a [separate ticket ](https://app.asana.com/0/1200180014510248/1203331355829273/f)on deck to address it!

Reply:
Ah, perfect! Apologies if we had already talked about that and I just forgot.